src/sdk/namespace_fs.js

```diff
@@ -514,7 +517,36 @@ class NamespaceFS {
             await this._load_bucket(params, fs_context);
             const stat = await nb_native().fs.stat(fs_context, file_path);
             if (isDirectory(stat)) throw Object.assign(new Error('NoSuchKey'), { code: 'ENOENT' });
-            return this._get_object_info(params.bucket, params.key, stat);
+
```

Comment:
Hi @v-harihar A great start! :)
this functionality of searching a version in .versions/ or fallback to the parent folder is needed for almost every API function.
In order to write and maintain this logic once, it should move out to a helper function outside of read_object_md.
I would create a function with similar functionality to what you did but I would leave the delete marker error handling to read_object_md/read_object_stream.
that way, instead of calling: `const file_path = this._get_file_path(params);` I would call a function that does this searching logic.
After that change, you will need to do the delete marker check logic once

Reply:
sure .. I will rebase this patch once your patch is merged.